### ISSUE TITLE
Retry conficuration count

### ISSUE THREAD

**Issue by kamen-hursev** [govdelivery/sidekiq-retries#6]
Is this [msg['retry'] = e.max_retries || '1'](https://github.com/govdelivery/sidekiq-retries/blob/master/lib/sidekiq/retries/server/middleware.rb#L16) really correct? When you pass `'1'` as a String sidekiq always uses the default configuration option. (Because of this method [retry_attempts_from(msg_retry, default)](https://github.com/mperham/sidekiq/blob/c97b9ea72390637434b5ee8c8101ab07ddd2d1b5/lib/sidekiq/middleware/server/retry_jobs.rb#L165))

So in a case like this:
```
class SomeWorker
  include Sidekiq::Worker
  sidekiq_options retry: 10

  sidekiq_retries_exhausted do |msg|
    ...
  end

  def perform(args)
    fail Sidekiq::Retries::Retry.new(some_error)  
  end
end
```
We'll hit `sidekiq_retries_exhausted` block after the default 25 retries and not after 10.

PS: Is this project still supported?

**Comment by substars** [comment]
I actually removed documentation for that max_retries bit (556491f) because we weren't using it and the use case was kind of questionable. So yeah, this definitely looks wrong, but I think that `|| '1'` probably just shouldn't be there (e.g. the default should be the default unless you specify otherwise).

That said, my reading of max_retries is that it should use the your worker's default (`@max_retries`, which should be 10) the way the code is now. Not sure what's causing that not to happen.

https://github.com/mperham/sidekiq/blob/c97b9ea72390637434b5ee8c8101ab07ddd2d1b5/lib/sidekiq/middleware/server/retry_jobs.rb#L89

Re: support, we do use this gem with the latest Sidekiq (though we tend to prefer using Sidekiq::Retries::Fail for simplicity). Always happy to see PRs, otherwise will review issues as time allows.

**Comment by kamen-hursev** [comment]
Yes it seems that @max_retries, should be 10, but for some reason it's not... (or I have "miss-debugged" something). For now I have this worked around: `fail Sidekiq::Retries::Retry.new(some_error, 10)`
